### ISSUE TITLE
bug: anchor tags inside a button isn't a best practice and violates accessibility requirements

### ISSUE THREAD

**Issue by Chityalaakhil** [code100x/cms#1354]
**Describe the bug**
I'm seeing the usage of anchor tags inside a button element at multiple places throughout the site. Two interactive elements should not be nested.
Eg. Join discord link, Home, bookmarks links on the left nav in desktop, etc.
https://dequeuniversity.com/rules/axe/4.10/nested-interactive

**To Reproduce**
1. Head to https://app.100xdevs.com
2. Use keyboard to navigate the site (using tab).
3. All the links that were inside a button element receives focus twice (once to the button and next moves to inner anchor tag).
4. Inspect the element and notice the markup with anchor tag inside a button element.

**Expected behavior**
1. All the buttons with anchor tags inside should be modified to have either anchors or buttons as appropriate. Refer to this article https://frontendcoding.com/button-or-anchor/
2. Existing functionality should not break.

**Screenshots or GIFs**
![image](https://github.com/user-attachments/assets/2c7aa039-5b33-42d9-8556-c9ce86343672)

**Info (please complete the following information):**
- Browser [e.g. Edge ]
- Version [e.g. 129.0.2792.65]

**Comment by Jatin123lodhi** [comment]
Can you assign this to me, I figured out the solution. In the given image you can check I have removed the button wrapper over the anchor tag and now we only have to press tab once, similarly I have figured out for the discord btn
![Screenshot 2024-09-30 202625](https://github.com/user-attachments/assets/e7eef7e2-2d84-4e62-b4af-7d94852214f9)

**Comment by Chityalaakhil** [comment]
Please feel free to work on it.

**Comment by Jatin123lodhi** [comment]
should I raise pr or should I wait before this issue get assigned to me

**Comment by JaskiratAnand** [comment]
Fixed the issue in the courses section. The issue in the sidebar seems to be a part of the TooltipTrigger being used from the UI library "@radix-ui/react-tooltip"

https://github.com/user-attachments/assets/a0c5f94c-b4fb-4648-971e-885ce207794b

![Screenshot (440)](https://github.com/user-attachments/assets/3d014169-3021-48f9-9320-1245696839d5)

![Screenshot (442)](https://github.com/user-attachments/assets/e8114c3e-3e47-4a6b-9350-8e2a5daf4f29)

**Comment by Chityalaakhil** [comment]
@JaskiratAnand can u do a PR for the things that u was able to fix. I'll look into radix ui tooltip.

**Comment by Chityalaakhil** [comment]
> should I raise pr or should I wait before this issue get assigned to me @Chityalaakhil

@Jatin123lodhi I'm not able to assign issue and I don't think maintainers also assigning them. If u r looking into the fix, probably just add a comment and start if no one started it already.

**Comment by JaskiratAnand** [comment]
@Chityalaakhil When i commit changes to my forked branch, it is lining the entire codebase. 

![Screenshot (443)](https://github.com/user-attachments/assets/39cf58a8-8874-43ce-aa00-90fe252a17ec)

![Screenshot (444)](https://github.com/user-attachments/assets/4355a407-1e45-47f7-b3ad-35b48453f16f)

![Screenshot (445)](https://github.com/user-attachments/assets/208af137-b921-46a7-910a-80b3e454810c)

**Comment by Chityalaakhil** [comment]
@JaskiratAnand try to comment out this file for now. These formatting rules might have been added recently!
![image](https://github.com/user-attachments/assets/4eac10a4-2c68-4e60-b74b-bff7c6c46027)

**Comment by Chityalaakhil** [comment]
Above two PRs should fix issues mentioned.
